### PR TITLE
Export results in SARIF format

### LINKED ISSUE #1722: Results Output as SARIF format to integrate with Github
**Is your feature request related to a problem? Please describe.**
Kics should be able to generate the output of results in a SARIF format, so it can be quickly integrated with Github. 
kics-github-action will benefit from this.

**Describe the solution you'd like**
Additional Flag to generate SARIF format output file

**Describe alternatives you've considered**
CxFlow has this implemented already, and many other tools are integrating with Github using this format

### PR DESCRIPTION
**Pull request by felipe-avelar**
Closes #1722 

**Proposed Changes**

- Export results in SARIF format following OASIS defined standard
I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: pkg/model/sarif_test.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+// TestCreateSarifReport tests if creates a sarif report correctly
+func TestCreateSarifReport(t *testing.T) {
+	sarif := NewSarifReport().(*sarifReport)
+	require.Equal(t, "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json", sarif.Schema)
+	require.Equal(t, "2.1.0", sarif.SarifVersion)
+	require.Equal(t, "KICS", sarif.Runs[0].Tool.Driver.ToolName)
+	require.Equal(t, "https://www.kics.io/", sarif.Runs[0].Tool.Driver.ToolURI)
+	require.Equal(t, "Keeping Infrastructure as Code Secure", sarif.Runs[0].Tool.Driver.ToolFullName)
+	require.Equal(t, "1.1.2", sarif.Runs[0].Tool.Driver.ToolVersion)
```
Comment (ruigomescx) [inline pkg/model/sarif_test.go:17]: is this KICS version?

File: docs/configuration-file.md [hunk 2]
```diff
@@ -78,7 +79,8 @@ log-file: true
 no-color: false
 minimal: false
 no-progress: false
-output-path: "file path to store result in json format"
+output-formats: "formats the result will be exported"
```
Comment (ruigomescx) [inline docs/configuration-file.md:82]: maybe rephrase it to -> _formats in which the result will be exported_

File: docs/getting-started.md [hunk 3]
```diff
@@ -87,8 +87,13 @@ Flags:
       --minimal                   minimal version of results output
       --no-color                  disable color output
       --no-progress               hides the progress bar
+<<<<<<< HEAD
+  -o, --output-path string        directory path to store result in output formats
+      --output-formats string     formats the result will be exported
+=======
       --output-lines int          number of lines to be displayed in results output (default 3)
   -o, --output-path string        file path to store result in json format
+>>>>>>> master
```
Comment (ruigomescx) [inline docs/getting-started.md]: fix this merge leftovers

File: docs/sarif.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+### GitHub Actions integration with SARIF
+
+KICS has an option to generate results on SARIF formats, which can be integrated with many CI tools, [including GitHub Actions](https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/integrating-with-code-scanning).
```
Comment (ruigomescx) [inline docs/sarif.md:3]: formats -> format

File: docs/sarif.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      path: 'terraform'
+      output-formats: 'sarif'
+      output-path: 'results.sarif'
+  - name: Upload SARIF file
+    uses: github/codeql-action/upload-sarif@v1
+    with:
+      sarif_file: results.sarif
+```
+
+The results list can be found on security tab of your GitHub project and should loos like something as following:
```
Comment (ruigomescx) [inline docs/sarif.md:22]: loos... -> look like the following image:

File: docs/sarif.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+<img src="https://raw.githubusercontent.com/Checkmarx/kics/master/docs/img/sarif-example-1.png" width="850">
+
+And an entry should describe the error and which line it happened like following image:
```
Comment (ruigomescx) [inline docs/sarif.md:26]: "An entry should describe the error and in which line it occurred:"

File: internal/console/scan.go [hunk 7]
```diff
@@ -161,7 +162,8 @@ func initScanCmd() {
 		"./assets/queries",
 		"path to directory with queries",
 	)
-	scanCmd.Flags().StringVarP(&outputPath, "output-path", "o", "", "file path to store result in json format")
+	scanCmd.Flags().StringVarP(&outputPath, "output-path", "o", "", "directory path to store result in output formats")
+	scanCmd.Flags().StringSliceVarP(&outputFormats, "output-formats", "", []string{}, "formats the result will be exported")
```
Comment (ruigomescx) [inline internal/console/scan.go:166]: change the text here with recommendation done earlier

File: pkg/model/sarif.go [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+	DefinitionShortDescription sarifMessage `json:"shortDescription"`
+	DefinitionFullDescription  sarifMessage `json:"fullDescription"`
+}
+
+type sarifTaxonomy struct {
+	TaxonomyGUID             string                    `json:"guid"`
+	TaxonomyName             string                    `json:"name"`
+	TaxonomyFullDescription  sarifMessage              `json:"fullDescription"`
+	TaxonomyShortDescription sarifMessage              `json:"shortDescription"`
+	TaxonomyDefinitions      []sarifTaxanomyDefinition `json:"taxa"`
```
Comment (ruigomescx) [inline pkg/model/sarif.go:119]: why taxa name?

Comment (felipe-avelar) [reply pkg/model/sarif.go:119]: This is a definition for taxonomies object as you can see [here](https://docs.oasis-open.org/sarif/sarif/v2.1.0/cs01/sarif-v2.1.0-cs01.html#_Toc16012492)

**Pull request merged**